### PR TITLE
KeeperClient auto-detection of keeper node from clickhouse configs

### PR DESCRIPTION
**Pull request by pufit**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
If no hostname or port were specified, keeper client will try to search for a connection string in the ClickHouse's config.xml

https://github.com/ClickHouse/ClickHouse/issues/53591

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
This is an automated comment for commit 8c1f0251e6fb4925dc2ad6b3fd1b28bad229bd3b with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/53769/8c1f0251e6fb4925dc2ad6b3fd1b28bad229bd3b/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: programs/keeper-client/KeeperClient.cpp [hunk 1]
```diff
@@ -211,7 +217,14 @@ void KeeperClient::initialize(Poco::Util::Application & /* self */)
...
 
-    Poco::Logger::root().setLevel(config().getString("log-level", "error"));
+    String default_log_level;
+    if (config().has("query"))
+        /// We don't want to see any information log in query mode, unless it was set explicitly
+        default_log_level = "error";
+    else
+        default_log_level = "information";
+
+    Poco::Logger::root().setLevel( config().getString("log-level", default_log_level));
```
Comment (antonio2368) [inline programs/keeper-client/KeeperClient.cpp:227]:
```suggestion
    Poco::Logger::root().setLevel(config().getString("log-level", default_log_level));
```

File: programs/keeper-client/KeeperClient.cpp [hunk 2]
```diff
@@ -311,8 +324,31 @@ int KeeperClient::main(const std::vector<String> & /* args */)
...
-    auto port = config().getString("port", "9181");
+    DB::ConfigProcessor config_processor(config().getString("config-file", "config.xml"));
+
+    /// This will handle a situation when clickhouse is running on the embedded config, but config.d folder is also present.
+    config_processor.registerEmbeddedConfig("config.xml", "<clickhouse/>");
+    auto clickhouse_config = config_processor.loadConfig();
+
+    String host;
+    String port;
+    String prefix = "zookeeper.node[0].";
```
Comment (antonio2368) [inline programs/keeper-client/KeeperClient.cpp:335]: If I understand correctly, you add only the first connection string here, why wouldn't you add them all and let the client connect to one of them?

**Review by antonio2368 — APPROVED** [review]
Is it possible to write integration tests for it?

**Pull request merged**